src/github_runner_image_builder/errors.py

```diff
@@ -0,0 +1,89 @@
+# Copyright 2024 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Module containing error definitions."""
+
+
+class ImageBuilderBaseError(Exception):
```

Comment:
As above, defining a module for exceptions lowers cohesion

Reply:
This was discussed with @gregory-schiano as well - we've decided to keep a separate module in the end, with cohesive code being an option as well, but a preferential decision to keep it like this (libs like openstack also keeps an exceptions module separate)

Reply:
I'm ok with that, but the fact that this is common practice across the python community doesn't make it more correct

Reply:
Yup - I think this could be something we could discuss and agree on as a team :D Thank you for bringing it up!